app/react/Library/components/TableRow.tsx

```diff
@@ -0,0 +1,170 @@
+      break;
+    case 'geolocation':
+      result = <GeolocationViewer points={prop.value} onlyForCards />;
+      break;
+    default:
+      result = undefined;
+      break;
+  }
+  return result;
+}
+
+function formatDocument(document: any, templates: TemplateSchema[], thesauris: ThesaurusSchema[]) {
+  let formattedDoc = document.toJS();
+  const template = templates.find((t: TemplateSchema) => t.get('_id') === formattedDoc.template);
+  formattedDoc.templateName = template?.get('name');
+  formattedDoc = formatter.prepareMetadata(formattedDoc, templates, thesauris, null, {
+    sortedProperty: 'creationDate',
+  });
+  formattedDoc.metadata.forEach((prop: any) => {
+    formattedDoc[prop.name] = prop;
+  });
+  return formattedDoc;
+}
+
+function displayCell(
+  document: any,
+  column: any,
+  index: number,
+  firstColumnCheckbox: any,
+  zoomLevel: number
+) {
+  const property = document[column.get('name')];
+  const cellValue = formatProperty(property);
+
+  const otherColumn = (value: any) => (
+    <div className={`table-view-cell table-view-row-zoom-${zoomLevel}`}>{value}</div>
+  );
+  const firstColumn = (value: any) => (
+    <div>
+      {firstColumnCheckbox(index)}
+      {otherColumn(value)}
+    </div>
+  );
+
+  return (
+    <td className={!index ? 'sticky-col' : ''} key={index}>
+      {!index ? firstColumn(cellValue) : otherColumn(cellValue)}
+    </td>
+  );
+}
+
+class TableRowComponent extends Component<TableRowProps> {
+  constructor(props: TableRowProps) {
+    super(props);
+    this.onClick = this.onClick.bind(this);
+    this.firstColumnCheckbox = this.firstColumnCheckbox.bind(this);
+  }
+
+  onClick(e: { preventDefault: () => void }) {
+    if (this.props.onClick) {
+      this.props.onClick(e, this.props.document, this.props.selected);
+    }
+  }
+
+  firstColumnCheckbox(index: number) {
+    return (
+      !index && (
+        <input type="checkbox" checked={this.props.selected} onClick={this.onClick.bind(this)} />
+      )
+    );
+  }
+
+  render() {
+    const formattedDocument = formatDocument(
+      this.props.document,
+      this.props.templates,
+      this.props.thesauris
+    );
+
+    return (
+      <tr className={this.props.selected ? 'selected' : ''}>
+        {this.props.columns.map((column: any, index: number) =>
+          displayCell(
+            formattedDocument,
+            column,
+            index,
+            this.firstColumnCheckbox,
+            this.props.zoomLevel
+          )
+        )}
+      </tr>
+    );
+  }
+}
+
+function mapStateToProps(state: any, ownProps: TableRowProps) {
+  const selected = ownProps.storeKey
+    ? !!state[ownProps.storeKey].ui
+        .get('selectedDocuments')
+        .find((doc: any) => doc.get('_id') === ownProps.document.get('_id'))
+    : false;
+  return {
+    selected,
+    document: ownProps.document,
+    columns: ownProps.columns,
+    templates: state.templates,
+    thesauris: state.thesauris,
+    onClick: ownProps.onClick,
```

Comment:
I am confused here: mapStateToProps extends the passed props (ownProps) with the return props.

If the Component already has document, columns, and onClick props, you don't need to return them again here from ownProps.  They are already passed.  The only reason to do this would be to map to different named props, which is not the case.

Reply:
Yeah, agree